backend/open_webui/routers/knowledge.py

```
@@ -544,11 +571,19 @@ def remove_file_from_knowledge_by_id(
+
     # Remove content from the vector database
     try:
         VECTOR_DB_CLIENT.delete(
             collection_name=knowledge.id, filter={"file_id": form_data.file_id}
-        )
+        )  # Remove by file_id first
+
+        VECTOR_DB_CLIENT.delete(
+            collection_name=knowledge.id, filter={"hash": file.hash}
+        )  # Remove by hash as well in case of duplicates
     except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)